### PR TITLE
Alleny244 patch 1

### PR DESCRIPTION
**Pull request by Alleny244**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)
Yes
### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)
Yes

### CODE REVIEW

File: Java/ds/README.md [hunk 1]
```diff
@@ -1 +1,2 @@
-a
+
+*  [Sparse Matrix](SparseMatrix.java)
```
Comment (HarshCasper) [inline Java/ds/README.md]: Don't make the README updates here

File: Java/ds/SparseMatrices.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                    sparseMatrix[k][1] = j;
+                    sparseMatrix[k][2] = inputMatrix[i][j];
+                    k++;
+                    // count variable keeps a track on non-zero entries in the given matri
+                    count += 1;
+                }
+            }
+            // First row's third column denotes
+            // number of non - zero elements in
+            // sparseMatrix
```
Comment (HarshCasper) [inline Java/ds/SparseMatrices.java]: Use Multi-Line Comments

File: Java/ds/SparseMatrices.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
+//          SPARSE MATRIX
+
+
+//  Java program for Sparse Matrix Representation using Array
```
Comment (HarshCasper) [inline Java/ds/SparseMatrices.java]: Add more description

File: Java/ds/SparseMatrices.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
+//          SPARSE MATRIX
+
+
+//  Java program for Sparse Matrix Representation using Array 
+import java.util.*;
+
+public class SparseMatrices{
+
+    public static void main(String[] args) {
```
Comment (HarshCasper) [inline Java/ds/SparseMatrices.java]: Use main() function only for the driver code. Use a separate function for putting all your logic

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: Java/ds/README.md (hunk 1 above)
Comment (Alleny244) [reply Java/ds/README.md]: I have deletd that and updated the correct readme.md file

File: Java/ds/SparseMatrices.java (hunk 2 above)
Comment (Alleny244) [reply Java/ds/SparseMatrices.java]: I have changed it into Multi-Line Comments

File: Java/ds/SparseMatrices.java (hunk 3 above)
Comment (Alleny244) [reply Java/ds/SparseMatrices.java]: I have added more description

File: Java/ds/SparseMatrices.java (hunk 4 above)
Comment (Alleny244) [reply Java/ds/SparseMatrices.java]: Changed as per the requirments

**Comment by Alleny244** [comment]
I have made the pull request after making all the changes

**Comment by Alleny244** [comment]
@HarshCasper can you review it again

**Review by HarshCasper — APPROVED** [review]

**Review by ankitaggarwal23 — COMMENTED** [review]
Problem already present `- [Sparse Matrix](ds/SparseMatrix.java)`

**Pull request merged**